gateway/ln-gateway/src/lib.rs

```diff
@@ -1599,14 +1606,24 @@ impl Gateway {
     pub async fn handle_shutdown_msg(&self, task_group: TaskGroup) -> AdminResult<()> {
         // Take the write lock on the state so that no additional payments are processed
         let mut state_guard = self.state.write().await;
-        if let GatewayState::Running { lightning_context } = state_guard.clone() {
+
+        // Take the current state, replacing it with a dummy variant.
+        // We'll restore the original state if the shutdown process fails.
+        // This is safe since we have the write lock on the state.
+        let old_state = std::mem::replace(&mut *state_guard, GatewayState::Disconnected);
```

Comment:
Why are we making this change in this PR? What is the purpose of this change?

Reply:
Whoops! That was from a separate tinkering session. Not sure how that ended up in here.